test/BUILD.bazel

```diff
@@ -71,6 +71,8 @@ cc_library(
         "include/test/test.h",
         "include/test/thd.h",
         "include/test/timer.h",
+        "//test/unit:arena_reset.c",  # included by arena_reset_prof.c
```

Comment:
The junk tests also include sources, I think we can be faithful to the make build without exposing these to everything that depends on testlib. I added library targets that scoped to the unit package.